app/database/attempt_store_compute_all_attempts.go

```diff
@@ -57,7 +57,20 @@ func (s *AttemptStore) ComputeAllAttempts() (err error) {
 				WHERE (
 					descendants.result_propagation_state = 'to_be_recomputed' OR
 					descendants.result_propagation_state = 'changed'
-				) AND existing.id IS NULL
+				) AND existing.id IS NULL AND EXISTS(
+					SELECT 1 FROM permissions_generated
+					JOIN groups_ancestors_active
+						ON groups_ancestors_active.ancestor_group_id = permissions_generated.group_id
+					WHERE
+						(permissions_generated.item_id = items_ancestors.ancestor_item_id OR
+						 permissions_generated.item_id IN (
+								 SELECT grand_ancestors.ancestor_item_id
+								 FROM items_ancestors AS grand_ancestors
+								 WHERE grand_ancestors.child_item_id = items_ancestors.ancestor_item_id
+						)) AND permissions_generated.can_view_generated != 'none' AND
```

Comment:
It is efficient (I checked this) and much more convenient because otherwise we would join same tables multiple times with different aliases which would make the query much more complex and much less readable.

Reply:
Fine then.
Yes, for readability it is almost always more readable, for sure. :-)